ruby-challenges.rb

```diff
@@ -2,29 +2,150 @@
+
+ 
+
+# (Final attempt/attempt 3) needs a refactor was unable to get an array. 😥
+us_states.keys.sort_by { |key| us_states[key] }.each do
+    |key|
+        puts us_states[key].sort
+    end
+
+    # output : Arizona
+# California
+# Nevada
+# Maine
+# New Hampshire
+# Rhode Island
+# Idaho
+# Oregon
+# Washington
+
+# (attempt 2) 
+# us_states.each do | key,value|
+    
+# end
+# print us_states.flatten.
+
+#(attempt1 ) us_states.each do 
+#     |value|
+# end
+# print us_states.map(&:last)
+
+
+
+
+# (attempt 1) us_states.each_with_index do | hash_index|
+#     puts us_states.flatten
+# end
+
+
+# --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object
+
+# Psuedocode: Create a class called Bike? what does this class do ? Classes are the basic building blocks in Object-Oriented Programming (OOP) & Ruby. Classes  help you define a blueprint for creating objects.# Objects are the products of the class. I want to use the Bike Class to be a container of methods, instance variables & constants, forming a blueprint which I can create objects from.
+# I want my Bike class to hold on to instances of model, wheels, and current_speed with specific numeric values. 
+# Let’s define a class Bike. Bike's have a model, have wheels, and have speed rates.
+
+#     Here’s how to define a  new, empty class Bike:
+    
+#     class Bike
+#     end
+    
+#     It's not a very useful class, but we can instantiate it, and create an actual, concrete Bike instance (object) from it:
+    
+#     p Bike.new
+#     Now, before I add any behaviour (methods) to the class, I want to be able to give it some initial data: In my case, I want the car to know its model type, current speed and wheels .
+#     I can do this like so:
+#     class Bike
+#       def initialize(model, wheels, current_speed)
+#       end
+#     end
+#     You see that we add a method called initialize to the class, and this method accepts a three argumens, which is called model, wheels and current_speed. At the moment, this method is still empty. I'll add some code to it. the method initialize is a special method with a special meaning in Ruby:
+#     Whenever you call the method new on a class, as in method.new, the class will create a new instance of itself. It will then, internally, call the method initialize on the new object. Doing so it will simply pass all the arguments that you passed to new on to the method initialize.   
+#     So we can now create a new Bike instance by calling …
+#     Bike.new("Trek")
+#     … and the string "Trek" will be passed on to our initialize method, and end up being assigned to the local variable .
+class Bike
+    attr_accessor :model, :wheels, :current_speed
+   def initialize(model,wheels,current_speed)
```

Comment:
Great !! Can we initialize it with a default 2 wheels and 0 mph for the current speed?